packages/astro/src/@types/astro.ts

```diff
@@ -2186,6 +2186,30 @@ export interface AstroUserConfig {
 		 */
 		serverIslands?: boolean;
 
+		/**
+		 * @docs
+		 * @name experimental.contentCollectionIntellisense
+		 * @type {boolean}
+		 * @default `false`
+		 * @version 4.14.0
+		 * @description
+		 *
+		 * Enables Intellisense features (e.g. code completion, quick hints) for your content collection entries in compatible editors.
+		 *
+		 * When enabled, this feature will generate and add JSON schemas to the `.astro` directory in your project. These files can be used by the Astro language server to provide Intellisense inside content files (`.md`, `.mdx`, `.mdoc`).
+		 *
+		 * ```js
+		 * {
+		 *   experimental: {
+		 *     contentIntellisense: true,
+		 *   },
+		 * }
+		 * ```
+		 *
+		 * To use this feature with the Astro VS Code extension, you must also enable the `astro.content-intellisense` option in your VS Code settings. For editors using the Astro language server directly, pass the `contentIntellisense: true` initialization parameter to enable this feature.
```

Comment:
If there *were* a link to a discussion or RFC where you want people to leave feedback, or where there's more helpful information, we'd typically include it at the end here. If not, then no worries! We don't have this for some of our longer-standing experimental flags.

Reply:
I don't expect it to be flagged for too long (it might get unflagged before 5.x), so I'd rather people report issues directly if there's bugs vs a discussion, I think